### PR TITLE
Brandon Johnson

### PR DESCRIPTION
**Pull request by brenicillin**
Feeling pretty solid about this one. The feedback I got on the paired project was a lot of help in writing my testing and making sure that everything is where it's supposed to be. I'm concerned that my extension using a .joins in the Supermarkets controller may violate MVC but would love to hear where it could be placed to better adhere. Overall, feeling good going into week 4!

### CODE REVIEW

File: spec/features/customer/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      expect(page).to_not have_content("Bread")
+
+      within '#add_items' do
+        expect(page).to have_field(:item_id)
+        expect(page).to have_button("Submit")
+        fill_in :item_id, with: @item_3.id
+        click_button "Submit"
+      end
+      
+      expect(page).to have_content("Bread")
```
Comment (cjsim89) [inline spec/features/customer/show_spec.rb:38]: Overall a good use of `within` blocks, and nice job using a helper method to set up your data elsewhere. I think for this particular test you could include another line to check the `current_path`, since the user story says that they're redirected somewhere (even if it is back to the page they were already on).

File: spec/features/supermarket/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  describe 'As a visitor' do
+    it 'I can see a list of all customers' do
+      visit "/supermarkets/#{@supermarket.id}"
+
+      within "#customer_#{@customer_1.id}" do
+        expect(page).to have_content(@customer_1.name)
+      end
+
+      within "#customer_#{@customer_2.id}" do
+        expect(page).to have_content(@customer_2.name)
```
Comment (cjsim89) [inline spec/features/supermarket/show_spec.rb:26]: The user story is asking for "a unique list", so our features will need to check that these names only appear once. To do this, you can add `, count: 1` to the end of your `have_content(...` parameters. I.e.
```
expect(page).to have_content(@customer_2.name, count: 1)
```

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -3,5 +3,19 @@
...
+  it 'can count customers' do
+    customer_1 = Customer.create!(name: "Bob")
+    customer_2 = Customer.create!(name: "Sally")
+    supermarket = Supermarket.create!(name: "Publix", location: "123 Main Street")
+    item = supermarket.items.create!(name: "Milk", price: 3)
+
+    customer_1.items << item
+    customer_2.items << item
+
+    expect(item.customer_count).to eq(2)
```
Comment (cjsim89) [inline spec/models/item_spec.rb:19]: This test is ok; I think a good edge case for this might be, what if a customer buys the same item twice? Would that be counted as 2 customers...?
